src/common/utils/NebulaKeyUtils.cpp

```diff
@@ -86,6 +86,92 @@ std::string NebulaKeyUtils::vertexKey(size_t vIdLen,
       .append(vIdLen - vId.size(), pad);
   return key;
 }
+
+// static
+std::string NebulaKeyUtils::vertexRefKey(size_t vIdLen,
+                                         PartitionID partId,
+                                         const VertexID& vId,
+                                         char pad) {
+  CHECK_GE(vIdLen, vId.size());
+  int32_t item = (partId << kPartitionOffset) | static_cast<uint32_t>(NebulaKeyType::kVertexRef);
+  std::string key;
+  key.reserve(sizeof(PartitionID) + vIdLen);
+  key.append(reinterpret_cast<const char*>(&item), sizeof(int32_t))
+      .append(vId.data(), vId.size())
+      .append(vIdLen - vId.size(), pad);
+  return key;
+}
+
+std::string NebulaKeyUtils::vertexRefVal(std::set<TagID> tagIds) {
+  std::string val;
+  apache::thrift::CompactSerializer::serialize(tagIds, &val);
+  return val;
+}
+
+std::set<TagID> NebulaKeyUtils::parseVertexRefVal(folly::StringPiece rawData) {
+  std::set<TagID> tagIds;
+  apache::thrift::CompactSerializer::deserialize(rawData, tagIds);
```

Comment:
for every read, do we have to do a deserialize every time?

Reply:
how fast is it ? copy from rocksdb plain table, deserialize, to the memory (set / map)? 100 us?